src/lib/QueryType.hs

```diff
@@ -385,6 +387,16 @@ projectNewtype ty = case ty of
   RecordTy _ -> throw CompilerErr "Can't project partially-known records"
   _ -> error $ "not a newtype: " ++ pprint ty
 
+-- adds newtypes (deeply) to a newtype-stripped atom
+wrapNewtypes :: EnvReader m => Type n -> Atom n -> m n (Atom n)
+wrapNewtypes ty x = do
+  case ty of
+    NatTy -> return $ Con $ Newtype NatTy x
+    _ -> return x -- TODO: handle other newtypes
```

Comment:
Perhaps `error` instead of causing type errors? :)

Reply:
I guess you do depend on it not failing for other types, but IMO this is too dangerous to put in a common place like this. Perhaps copy it over as a helper in export and leave a TODO to generalize and move to QueryType?

Reply:
Yes, good call